### PR TITLE
PT-BR : Add file what-is-the-arguments-object.md 

### PR DESCRIPTION
**Pull request by fernandoors**
Hello everyone,

The translation file 'what-is-the-arguments-object.md'  was completed. 

@nodejs/nodejs-pt 
@khaosdoctor

### CODE REVIEW

File: locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  - truthy
+  - falsy
+  - types
+  - coercion
+title: What is the arguments object?
+difficulty: 4
+layout: knowledge-post.hbs
+---
+
+Os objetos `arguments` são uma estrutura especial disponível dentro de todas as chamadas de função e que é representada como uma lista de argumentos que são passadas quando a função é chamada. Desde que JavaScript permite chamar funções com indeterminado número de argumentos, nós precisamos de um método para descobrir e acessar esses argumentos.
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md]:
```suggestion
O objeto `arguments` é uma estrutura especial disponível dentro de todas as chamadas de função e que é representada como uma lista de argumentos que são passados quando a função é chamada. Uma vez que o JavaScript permite chamar funções com número indeterminado de argumentos, nós precisamos de um método para descobrir e acessar esses argumentos.
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+Os objetos `arguments` são um tipo de objeto array, seu tamanho corresponde a quantidade de argumentos passados para a função e que é possível acessar seu valor através de indexação do Array. Exemplo `arguments[0]` //Captura ao primeiro argumento. Existe apenas mais uma propriedade de `arguments` chamada `callee`, Nota: o ES5 impede o uso do `strict mode` mais informações podem ser encontradas [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments/callee). Veja mais exemplos sobre as propriedades `arguments`.
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md]:
```suggestion
O objeto `arguments` é um array-like, seu tamanho corresponde a quantidade de argumentos passados para a função. É possível acessar estes valores através da indexação do Array. Exemplo: `arguments[0]` captura o primeiro argumento. A única outra propriedade de `arguments` é chamada `callee`, a qual o ES5 impede o uso no `strict mode`, mais informações podem ser encontradas [aqui](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments/callee). 

Veja mais exemplos sobre as propriedades de `arguments`.
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+```js
+const minhaFunc = function(primeiro) {
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md]: This code is different from the one in the docs

```
const myfunc = function(one) {
  arguments[0] === one;
  arguments[1] === 2;
  arguments.length === 3;
}

myfunc(1, 2, 3);
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  const index0 = arguments[0] === primeiro;
+  const index1 = arguments[1] === 2;
+  const argumentsLength = arguments.length === 3;
+  return console.log(index0, index1, argumentsLength)
+}
+minhaFunc(1, 2, 3);
+// expectativa de retorno no console: true | true | true
+```
+```js
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md]: This example doesn't exist in the document, if you'd please remove it :)

File: locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+const minhaFunc = function(primeiro) {
+  const index0 = arguments[0] === primeiro;
+  const index1 = arguments[1] === 2;
+  const argumentsLength = arguments.length === 3;
+  return console.log(index0, index1, argumentsLength)
+}
+minhaFunc(1, 2, 3);
+// expectativa de retorno no console: true | true | true
+```
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md]: In between those two examples, there is another paragraph which is missing:

*"This construct is very useful and gives javascript functions a lot of flexibility. But there is an important gotcha. The arguments object behaves like an array, but it is not an actual array. It does not have Array in its prototype chain and it does not respond to any array methods, e.g. arguments.sort() raises a TypeError. Instead, you need to copy the values into a true array first. With the advent of ES6 Array.from() method this is quite straightforward."*

Please add it so the docs are complete :)

File: locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  )
+}
+
+locais('casa', 'praia', 'fazenda');
+// expectativa de retorno no console: true | false | true
+```
+
+NOTA: Para ES5 ou anteriores, o loop `for` serve para coletar os dados dos `arguments`
+
+```js
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md]: This example does not exist in the docs if you could remove it please :)

File: locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+function casa(){
+  for (locais in arguments){
+    console.log(arguments[locais])
+  }
+  return arguments.length
+}
+casa('quarto', 'banheiro', 'sala')
+// expectativa de retorno no console: quarto | banheiro | sala | 3
+```
+Em alguns casos você ainda pode tratar o `arguments` como um Array, neste caso você pode usar o `arguments` através de funções dinâmicas. Os métodos nativos do Array (Ex. Array.prototype.concat) serão aceitos para tratar os `arguments` dinamicamente. Essa técnica também oferece outros modos de conversão dos `arguments` para um tipo Array utilizando do método `Array.slice`.
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md]:
```suggestion
Em alguns casos você ainda pode tratar o `arguments` como um Array, você pode usar o `arguments` através de invocações de funções dinâmicas. E a maioria dos métodos nativos do Array (Ex. Array.prototype.concat) aceitarão `arguments` quando invocados dinamicamente utilizando `call` ou `apply`. Essa técnica também oferece outros modos de conversão dos `arguments` para um tipo Array utilizando do método `Array.slice`.
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+```js
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md:31]: This whole example is different from the one in the docs:

**Original:**
```js
myfunc.apply(obj, arguments).

// concat arguments onto the 
Array.prototype.concat.apply([1,2,3], arguments);

// turn arguments into a true array
const args = Array.prototype.slice.call(arguments);

// cut out first argument
args = Array.prototype.slice.call(arguments, 1);
```

File: locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+const minhaFunc = function(primeiro) {
+  return console.log( Array.prototype.slice.call(arguments, 1))
+}
+
+minhaFunc(1,2,3,'PIN',5,6,7)
+// expectativa de retorno no console: [ 2, 3, 'PIN', 5, 6, 7 ]
+
+```
+
+O `arrow functions` foram adicionados no ECMAScript 2015 (ES6), propondo uma sintaxe mais compacta do que em sua versão anterior. Com isso, houve um modo alternativo para tratar os `arguments` o convertendo para `arguments objects` (além das palavras chaves `this`, `super`, e `new.target`). A solução para esse caso de uso é utilizar `rest parameter`. O `rest parameter` permite representar um indeterminado numero de argumentos como um tipo Array. Leia mais detalhes [aqui](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters).
```
Comment (khaosdoctor) [inline locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md]:
```suggestion
As `arrow functions` foram adicionadas na especificação do ECMAScript 2015 (ES6), propondo uma alternativa mais compacta do que a declaração de uma função comum. A desvantagem dessa alternativa é que não existe mais o objecto `arguments` (e também as keywords `this`, `super` ou `new.target`). A solução para esses casos é o uso dos parâmetros `rest`. Os parâmetros `rest` permitem representar um número indeterminado de argumentos como um Array. Leia mais detalhes [aqui](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters).
```

**Review by khaosdoctor — CHANGES_REQUESTED** [review]
Please do **not** add or remove any words or phrases from the docs, since it'd create a different documentation for each language. To localize and apply region-like expressions is ok, however, completely change phrases, add new paragraphs or remove others is something we can't do, since we'd create a hard fork of the original docs.

Also, please do not change, add or remove examples from the docs. Let's try to keep it as much closer to the original English docs as possible :)

**Comment by fernandoors** [comment]
> Please do **not** add or remove any words or phrases from the docs, since it'd create a different documentation for each language. To localize and apply region-like expressions is ok, however, completely change phrases, add new paragraphs or remove others is something we can't do, since we'd create a hard fork of the original docs.
> 
> Also, please do not change, add or remove examples from the docs. Let's try to keep it as much closer to the original English docs as possible :)

Hi @khaosdoctor thanks for any suggestions. I'll applying your recommendations in the next commit.

**Review by khaosdoctor — APPROVED** [review]
LGTM 👍

**Comment by khaosdoctor** [comment]
I think this one is good to go @MaledongGit

**Pull request merged**